src/event bound images.js

```diff
@@ -0,0 +1,82 @@
+5. As a brief explanation of the code:
+	- "SHOW_IMAGE" is a native bipsi method to display an image.  The last two parameters (both 0 in
+		the example) are used differently for an event-bound image, though.  For an event-bound
+		image they determine where the image is drawn relative to the event its bound to.
+	- "PLAYBACK.bindImageToEvent" is what makes the image move with the avatar.  It takes the image
+		id that was passed to "SHOW_IMAGE" and the event to bind the image to.
+*/
+
+// A method for the user to call, after calling SHOW_IMAGE, to bind the new image to an event.
+BipsiPlayback.prototype.bindImageToEvent = function bindImageToEvent(imageId, event) {
+	const image = this.images.get(imageId);
+	if (!image) {
+		this.showError(`Attempted to bind an unassigned image to an event: "${imageId}".`);
+		return;
+	}
+	// If there were no avatar-bound images up to now, start the frame updates.
+	if (!Object.keys(this.eventBoundImages).length) {
+		this.eventBoundImagesAnimationId = window.requestAnimationFrame(window.updateAllEventBoundImagePositions);
+	}
+	// Add the image/event combo to the binding list
+	this.eventBoundImages[imageId] = { event, image, offset: [image.x, image.y] };
+	// Update the new bound image's position immediately.  If we don't, it'll flash at the upper-left corner before the next update.
+	const currentRoomId = window.roomFromEvent(window.PLAYBACK.data, window.getEventById(window.PLAYBACK.data, window.PLAYBACK.avatarId)).id;
+	window.updateEventBoundImagePosition(this.eventBoundImages[imageId], currentRoomId);
+};
+
+// Modify PLAYBACK.hideImage - remove an image from the avatar-bound list, if it's there.
+wrap.after(BipsiPlayback.prototype, 'hideImage', imageId => {
+	delete window.PLAYBACK.eventBoundImages[imageId];
+	if (!Object.keys(window.PLAYBACK.eventBoundImages).length) {
+		window.cancelAnimationFrame(window.PLAYBACK.eventBoundImagesAnimationId);
+	}
+});
+
+// Add new fields to PLAYBACK.
+wrap.after(window, 'start', () => {
+	window.PLAYBACK.eventBoundImages = {};
+	window.PLAYBACK.eventBoundImagesAnimationId = null;
+});
+
+// Update positions for all avatar-bound images, based on their events' positions.
+window.updateAllEventBoundImagePositions = () => {
+	const currentRoomId = window.roomFromEvent(window.PLAYBACK.data, window.getEventById(window.PLAYBACK.data, window.PLAYBACK.avatarId)).id;
+	Object.values(window.PLAYBACK.eventBoundImages).forEach(binding => window.updateEventBoundImagePosition(binding, currentRoomId));
+	window.PLAYBACK.eventBoundImagesAnimationId = window.requestAnimationFrame(window.updateAllEventBoundImagePositions);
+};
+
+// Update the position for a single avatar-bound image, based on its event's position.
+window.updateEventBoundImagePosition = (binding, currentRoomId) => {
+	if (window.roomFromEvent(window.PLAYBACK.data, binding.event).id !== currentRoomId) {
+		binding.image.x = 999;
```

Comment:
this is hacky lol, i assume you're just trusting that 999 will be offscreen, but if that's the case:
- add a `// HACK: ` comment with that explanation
- use the actual bipsi map/tile size variables so it's guaranteed to be offscreen even if those change